action_server_bt/src/action_server_bt.cpp

```diff
@@ -119,30 +124,33 @@ void ActionServerBT::execute(const std::shared_ptr<GoalHandleActionTree> goal_ha
       feedback->node_status = convert_node_status(status);
       goal_handle->publish_feedback(feedback);
 
-      if (!loopRate.sleep())
-      {
-        RCLCPP_WARN(kLogger, "Behavior Tree tick frequency %d Hz was exceeded!",
-                    static_cast<int>(1.0 / (loopRate.period().count() * 1.0e-9)));
-      }
+      // sleep to tick the tree at the desired frequency
+      tree.sleep(std::chrono::milliseconds(static_cast<int>(1000.0 / params_.behavior_tick_frequency)));
     }
   }
   catch (const std::exception& ex)
   {
-    action_result->error_message = std::string("Behavior Tree exception: ") + ex.what();
+    action_result->error_message = "Behavior Tree exception: %s", ex.what();
```

Comment:
since when is this syntax possible?

Reply:
Ya, I was surprised to find some examples using this and wanted to try it out but forgot I left it in here. I can switch it back if you'd like this to look like standard c++ syntax.